src/FixedDataTable.js

```diff
@@ -1321,13 +1323,13 @@ var FixedDataTable = createReactClass({
       //NOTE (asif) This is a hacky workaround to prevent FDT from setting its internal state
       var onHorizontalScroll = this.props.onHorizontalScroll;
       if (onHorizontalScroll ? onHorizontalScroll(x) : true) {
-        this.setState({
+        newState = {
           scrollX: x,
-        });
+        };
       }
     }
 
-    this._didScrollStop();
+    this.setState(newState, this._didScrollStop);
```

Comment:
same logic as explained above: call `didScrollStop` only after `this.state` is updated